compiler/ilgen/IlBuilder.cpp

```diff
@@ -516,29 +517,33 @@ IlBuilder::appendBlock(TR::Block *newBlock, bool addEdge)
    _currentBlock = newBlock;
    }
 
-
 void
-IlBuilder::AppendBuilder(TR::IlBuilder *builder)
+IlBuilder::AppendBuilder(TR::IlBuilder *builder, bool addEdge)
    {
-   ILB_REPLAY("%s->AppendBuilder(%s);", REPLAY_BUILDER(this), REPLAY_BUILDER(builder));
+   ILB_REPLAY("%s->AppendBuilder(%s, %s)", REPLAY_BUILDER(this), REPLAY_BUILDER(builder), REPLAY_BOOL(addEdge));
    TR_ASSERT(builder->_partOfSequence == false, "builder cannot be in two places");
 
    builder->_partOfSequence = true;
    _sequenceAppender->add(builderEntry(builder));
-   if (_currentBlock != NULL)
+   if (_currentBlock != NULL && addEdge)
       {
       cfg()->addEdge(_currentBlock, builder->getEntry());
       _currentBlock = NULL;
       }
 
-   TraceIL("IlBuilder[ %p ]::AppendBuilder %p\n", this, builder);
+   TraceIL("IlBuilder[ %p ]::AppendBuilder %p addEdge %d\n", this, builder, addEdge);
 
    // when trees are connected, exit block will swing down to become last trees in builder, so it can fall through to this block we're about to create
    // need to add edge explicitly because of this exit block sleight of hand
    appendNoFallThroughBlock();
    cfg()->addEdge(builder->getExit(), _currentBlock);
    }
 
+void
+IlBuilder::appendExceptionBuilder(TR::IlBuilder *builder)
+   {
+   AppendBuilder(builder, false);
```

Comment:
Rather than add the boolean parameter to AppendBuilder, maybe the better approach is to:
TR::Block *newBlock = NULL;
if (_currentBlock != NULL)
   {
   newBlock = emptyBlock();
   Goto(thatBlock);
   }
AppendBuilder(handler);
if (newBlock == NULL)
   newBlock = emptyBlock();

appendBlock(newBlock, false);


That way, the service is self contained and you don't need to add an optional parameter to the public AppendBuilder() API.